pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -166,36 +170,72 @@ func generateMetadataStringMatcher(keys []string, v *metadata.StringMatcher) *me
 	return &metadata.MetadataMatcher{
 		Filter: authn.AuthnFilterName,
 		Path:   paths,
-		Value: &metadata.MetadataMatcher_Value{
-			MatchPattern: &metadata.MetadataMatcher_Value_StringMatch{
+		Value: &metadata.ValueMatcher{
+			MatchPattern: &metadata.ValueMatcher_StringMatch{
 				StringMatch: v,
 			},
 		},
 	}
 }
 
-// createDynamicMetadataMatcher creates a MetadataMatcher for the given key, value pair.
-func createDynamicMetadataMatcher(k, v string) *metadata.MetadataMatcher {
-	var keys []string
+func generateMetadataListMatcher(keys []string, v string) *metadata.MetadataMatcher {
 	forceRegexPattern := false
+	listMatcher := &metadata.ListMatcher{
+		MatchPattern: &metadata.ListMatcher_OneOf{
+			OneOf: &metadata.ValueMatcher{
+				MatchPattern: &metadata.ValueMatcher_StringMatch{
+					StringMatch: createStringMatcher(v, forceRegexPattern),
```

Comment:
You can just pass `false` here, no need for `forceRegexPattern`

Reply:
Done.